src/commands/deploy/index.js

```diff
@@ -144,7 +144,7 @@ export default async ctx => {
     platformVersion = versionFlag;
   }
 
-  if (platformVersion === 1 && versionFlag !== 1) {
+  if (platformVersion === 1 && versionFlag !== 1 && !argv['--docker'] && !argv['--npm']) {
```

Comment:
This looks correct. But why all the other changes?

Reply:
@styfle because due to the inaccurate lambda caching I've noticed that the `getTokenInformation` call had an incorrect retry call and therefore I changed it and simplified the other retry logic as well